ned/instance.py

```diff
@@ -150,42 +151,66 @@ def _ec2_config(
     return definition
 
 
-class Barrier:
+class InstanceBarrier:
     def wait(self, inst: Instance) -> Instance:
         return inst
 
 
 @dataclass(frozen=True)
-class InstanceRunningBarrier(Barrier):
+class InstanceReadyBarrier(InstanceBarrier):
     ec2: botocore.client.BaseClient
-    sleep_time: float = 10
-    retry_count: int = 9
+    sleep_time: float = 5
+    retry_count: int = 20
+
+    def _is_ready(self, inst: Instance) -> bool:
+        if inst.state != "running":
+            self._warn(f'Instance state is "{inst.state}", waiting for "running"')
+            return False
+
+        try:
+            resolved_ip = socket.gethostbyname(inst.public_dns_name)
+        except socket.gaierror as exc:
+            self._warn(f"Host lookup error: {exc}")
+            return False
+
+        if inst.public_ip_address != resolved_ip:
+            self._warn(
+                f"Instance resolved IP is {resolved_ip}, " f"but should be {inst.public_ip_address}"
+            )
+            return False
+
+        if not self._ping(inst.public_ip_address):
```

Comment:
Check if we can ping `inst.public_ip_address` (now same as `inst.public_dns_name`). If not, the instance is not ready.

This is where the failures in live testing usually occur, which leads to a wait and retry.

Reply:
neat!